diagram.md

```diff
@@ -2,240 +2,284 @@
-                    </tr><tr><td bgcolor="#CCFFFF">Timecode</td> 
-                      <td> </td> 
-                    </tr><tr><td bgcolor="#CCFFFF">Flags</td> 
-                      <td><table width="100%" border="0"><tr><td bgcolor="#CCFFFF">Gap</td> 
-                          </tr><tr><td bgcolor="#CCFFFF">Lacing</td> 
-                          </tr><tr><td bgcolor="#CCFFFF">Reserved</td> 
-                          </tr></table></td> 
-                    </tr></table></td> 
-              </tr></table></td> 
-        </tr><tr><td><table bgcolor="#FF9999" width="100%" border="0"><tr><td width="110" bgcolor="#FFCC99">Optional</td> 
-                <td><table width="100%" border="0"><tr><td bgcolor="#FFCC99">FrameSize</td> 
-                      <td> </td> 
-                    </tr></table></td> 
-              </tr></table></td> 
-        </tr><tr><td><table bgcolor="#9999FF" width="100%" border="0"><tr><td width="110" bgcolor="#99CCFF">Data</td> 
-                <td><table width="100%" border="0"><tr><td width="100" bgcolor="#99CCFF">Frame</td> 
-                      <td> </td> 
-                    </tr></table></td> 
-              </tr></table></td> 
-        </tr></table></td> 
-  </tr></table></div>
-<div align="center">Figure 3 </div> 
-
-Here is a representation of the Block structure. There is an in-depth discussion of it in the specs. I will add some descriptions here when I have time.
-
-One thing that I do want to mention however, to avoid confusion, is the Timecode. The quick eye will notice that there is one Timecode shown per Cluster, and then another within the Block structure itself. The way this works is that the Timecode in the Cluster is relative to the whole file. It is usually the Timecode that the first Block in the Cluster needs to be played at. The Timecode in the Block itself is relative to the Timecode in the Cluster. For example, let's say that the Timecode in the Cluster is set to 10 seconds, and you have a Block in that Cluster that is supposed to be played 12 seconds into the clip; this means that the Timecode in the Block would be set to 2 seconds.
- 
+# Matroska Structure
+
+A Matroska file is composed of one or many `EBML Documents` that use the `Matroska Document Type`. Each `EBML Document` MUST start with an `EBML Header` and then the `Root Element`, which is called `Segment` in Matroska. Matroska defines several `Top Level Elements` which MAY occur within the `Segment`.
+
+As an example, a simple Matroska file consisting of a single `EBML Document` could be represented like this:
+
+- EBML Header
+- Segment
+
+A more complex Matroska file consisting of an `EBML Stream`  (consisting of two `EBML Documents`) could be represented like this:
+
+- EBML Header
+- Segment
+- EBML Header
+- Segment
+
+The following diagram represents a simple Matroska file, comprised of an `EBML Document` with an `EBML Header`, a `Segment Element` (the `Root Element`), and all eight Matroska `Top Level Elements`.
```

Comment:
Maybe add some explanation that horizontal spacing expresses parent-child relationship/containment (e.g. the `Info` element is contained within the `Segment` element) whereas vertical alignment indicates the order in the file.